src/top/tensor/state_op.cc

```diff
@@ -0,0 +1,36 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file state_op.cc
+ * \brief State operators
+ */
+#include <nnvm/op.h>
+#include <nnvm/node.h>
+#include <nnvm/op_attr_types.h>
+#include <nnvm/top/tensor.h>
+#include "../op_common.h"
+#include "../elemwise_op_common.h"
+
+namespace nnvm {
+namespace top {
+
+NNVM_REGISTER_OP(assign)
```

Comment:
OK, stateful ops requires consideration and the current support may not be perfect. Let us first change the name to ```_assign``` for now